packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -718,20 +681,58 @@ class _GestureSemantics extends SingleChildRenderObjectWidget {
   }
 
   @override
+  Widget build(BuildContext context) {
+    Widget result = new Listener(
+      onPointerDown: _handlePointerDown,
+      behavior: widget.behavior ?? _defaultBehavior,
+      child: widget.child
+    );
+    if (!widget.excludeFromSemantics)
+      result = new _GestureSemantics(owner: this, child: result);
+    return result;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder description) {
+    super.debugFillProperties(description);
+    if (_recognizers == null) {
+      description.add(new DiagnosticsNode.message('DISPOSED'));
+    } else {
+      final List<String> gestures = _recognizers.values.map<String>((GestureRecognizer recognizer) => recognizer.debugDescription).toList();
+      if (gestures.isEmpty)
+        gestures.add('<none>');
+      description.add(new IterableProperty<String>('gestures', gestures));
+      description.add(new IterableProperty<GestureRecognizer>('recognizers', _recognizers.values, hidden: true));
+    }
+    description.add(new EnumProperty<HitTestBehavior>('behavior', widget.behavior, defaultValue: null));
+  }
+}
+
+class _GestureSemantics extends SingleChildRenderObjectWidget {
+  const _GestureSemantics({
+    Key key,
+    Widget child,
+    this.owner
+  }) : super(key: key, child: child);
+
+  final RawGestureDetectorState owner;
+
+  @override
   RenderSemanticsGestureHandler createRenderObject(BuildContext context) {
     final RenderSemanticsGestureHandler result = new RenderSemanticsGestureHandler();
     updateRenderObject(context, result);
```

Comment:
Can you have this call _updateHandlers directly? Having it bounce through updateRenderObject is weird.